core-bundle/src/EventListener/DataContainer/StoreRecordPreviewListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @Callback(table="tl_article", target="config.ondelete")
+ * @Callback(table="tl_content", target="config.ondelete")
+ * @Callback(table="tl_page", target="config.ondelete")
+ * @Callback(table="tl_page", target="config.ondelete")
```

Comment:
Instead of using the annotation I think in this case the callbacks should be registered dynamically, e.g. via an `loadDataContainer` hook. As you said, it needs to be done for all deletable DCAs anyway.

Reply:
Yes, that is the first todo in the opening comment ;-)